examples/http-request-errorbudget.yaml

```diff
@@ -0,0 +1,37 @@
+recordingrule:
+- expr: |
+    sum(label_replace(increase(promhttp_metric_handler_requests_total{namespace="default",job="fooapp"}[30d]), "status_class", "${1}xx", "code", "([0-9])..")) by (status_class)
+  labels:
+    job: fooapp
+    namespace: default
+  record: status_class:promhttp_metric_handler_requests_total:increase30d
+- expr: |
+    (
+      0.001000
+    *
+      sum(status_class:promhttp_metric_handler_requests_total:increase30d)
+    )
+  labels:
+    job: fooapp
+    namespace: default
+  record: promhttp_metric_handler_requests_errorbudget_requests
+- expr: |
+    (
+      sum(promhttp_metric_handler_requests_errorbudget_requests{namespace="default",job="fooapp"})
+    -
+      sum(status_class:promhttp_metric_handler_requests_total:increase30d{namespace="default",job="fooapp",status_class="5xx"})
+    )
+  labels:
+    job: fooapp
+    namespace: default
```

Comment:
If you do the 2nd sum `without (status_class)`, you don't need the 1st sum at all, and you'll naturally get the labels attached and don't need the explicit addition in the separate section.

Reply:
While that is true, I think the "stupid" and simple way of subtracting the two `sum` here is fine. We want to explicitly add the labels here again for the same reasons stated above. Very nice alternative though! :+1:

Reply:
Again, it throws away all the labels. Often, users want to propagate labels. As hinted at above, it would be quite an effort (and require an additional config option) to rework everything to perfectly propagate labels and at the same time allow regexp and negative selectors. Your call if you want to go down that road.